python-client/tests/communications/test_dto_serialization.py

```diff
@@ -0,0 +1,1283 @@
+    test_suite_dict_without_none = test_suite.dict(exclude_none=True)
+    for field in TEST_SUITE_NON_MANDATORY_FIELDS:
+        assert field in test_suite_dict_without_none.keys()
+
+
+TEST_ECHO_MSG_HB = "hb"
+
+
+def test_echo_message():
+    with pytest.raises(ValidationError):
+        websocket.EchoMsg()
+    echo_msg_dict = websocket.EchoMsg(msg=TEST_ECHO_MSG_HB).dict()
+    assert "msg" in echo_msg_dict.keys()
+    assert echo_msg_dict["msg"] == TEST_ECHO_MSG_HB
+
+
+def test_data_row():
+    with pytest.raises(ValidationError):
+        websocket.DataRow()
+
+    data_row_dict = websocket.DataRow(columns={}).dict()
+    assert "columns" in data_row_dict.keys()
+
+
+def test_data_frame():
+    with pytest.raises(ValidationError):
+        websocket.DataFrame()
+
+    data_frame_dict = websocket.DataFrame(rows=[]).dict()
+    assert "rows" in data_frame_dict.keys()
+
+
+def test_run_model_for_data_frame():
+    # Used by classification models
+    classification = websocket.RunModelForDataFrame(
+        all_predictions=websocket.DataFrame(
+            rows=[],
+        ),
+        prediction=[],
+    )
+    classification_dict = classification.dict()
+    assert "all_predictions" in classification_dict.keys()
+    assert "prediction" in classification_dict.keys()
+    assert "probabilities" in classification_dict.keys()
+    assert "raw_prediction" in classification_dict.keys()
+    classification_dict_without_none = classification.dict(exclude_none=True)
+    assert "all_predictions" in classification_dict_without_none.keys()
+    assert "prediction" in classification_dict_without_none.keys()
+    assert "probabilities" not in classification_dict_without_none.keys()
+    assert "raw_prediction" not in classification_dict_without_none.keys()
+
+    # Used by other models
+    other = websocket.RunModelForDataFrame(
+        prediction=[],
+        raw_prediction=[],
+    )
+    other_dict = other.dict()
+    assert "all_predictions" in other_dict.keys()
+    assert "prediction" in other_dict.keys()
+    assert "probabilities" in other_dict.keys()
+    assert "raw_prediction" in other_dict.keys()
+    other_dict_without_none = other.dict(exclude_none=True)
+    assert "all_predictions" not in other_dict_without_none.keys()
+    assert "prediction" in other_dict_without_none.keys()
+    assert "probabilities" not in other_dict_without_none.keys()
+    assert "raw_prediction" in other_dict_without_none.keys()
+
+
+TEST_PLATFORM_MACHINE = "arm64"
+TEST_PLATFORM_NODE = "Inokis-MacBook-Pro.local"
+TEST_PLATFORM_PROCESSOR = "arm"
+TEST_PLATFORM_RELEASE = "22.5.0"
+TEST_PLATFORM_SYSTEM = "Darwin"
+TEST_PLATFORM_VERSION = (
+    "Darwin Kernel Version 22.5.0: Thu Jun  8 22:21:34 PDT 2023; root:xnu-8796.121.3~7/RELEASE_ARM64_T8112"
+)
+
+
+def test_platform():
+    with pytest.raises(ValidationError):
+        websocket.Platform()
+    with pytest.raises(ValidationError):
+        websocket.Platform(
+            node=TEST_PLATFORM_NODE,
```

Comment:
It's too extensive, no need to test pydantic behaviour that fully.
One test maybe, not that many